### PR TITLE
connect api to MongoDB

### LINKED ISSUE #3106: [Dev] Connect API to mongodb
Install the following modules: `mongoose migrate-mongoose mongoose-patcher`. Create a file in api/db to connect to mongo using the MONGO_URL string. Require the file in api.js.

### This task is done when…
This could include:
 - [ ] install the following modules : `mongoose migrate-mongoose mongoose-patcher`
 - [ ] create a file in api/db to connect to the  
 - [ ] require the connection file in api.js

### PR DESCRIPTION
**Pull request by thetif**
resolves #3106 

**Description-**
Connect the api to the mongo server.

**This pull request changes...**

- created a mongo file that sets up the mongo connection
- required that file in the api.js so that it connects on start up
- set up migrate-mongoose so that we can handle migrations in mongo
- added migrate-mongoose scripts to package.json

**This pull request was tested in the follow ways…**

- using print statements on connection
- running migrate-mongoose create script from package.json and verified that migration files are created

**Steps to manually verify this change...**

- view logs from api and you should see the messages:
  - "Setting up MongoDB connection"
  - "MongoDB connected"

### This pull request is ready to review when...

- [ ] Changelog is updated as appropriate

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author

Code is being merged into an Epic branch and will be confirmed by Product and CMS when the entire Epic is completed. QA is not needed at this time because this is still a set up task.

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -15,6 +15,11 @@
     "seed": "knex seed:run --specific=main.js",
     "seed-dev": "NODE_ENV=development DEV_DB_HOST=localhost npm run seed",
     "seed-users": "knex seed:run --specific=base-users.js",
+    "make-mongoose-migrate": "migrate --config mongo-migrate.js create",
+    "make-mongoose-migrate-dev": "MONGO_URL=mongodb://mongo:cms@localhost:27017/eapd npm run make-mongoose-migrate",
+    "mongoose-migrate": "migrate --config mongo-migrate.js up",
+    "mongoose-migrate-dev": "MONGO_URL=mongodb://mongo:cms@localhost:27017/eapd npm run mongoose-migrate",
+    "mongoose-migrate-down": "migrate --config mongo-migrate.js down",
```
Comment (knollfear) [inline api/package.json:22]: The -dev version of this is missing.  However, I think the -dev versions are unnecessary.  How do you feel about updating the readme to indicate that someone should run ```export MONGO_URL=mongodb://mongo:cms@localhost:27017/eapd
``` before using these commands.  In addition is seems that for the create and down commands I need to add -- <NAME> to the command which wasn't super clear  maybe that can go in the readme too?

**Review by knollfear — COMMENTED** [review]
everything worked as expected.  I think maybe a bit more documentation is in order though.

File: api/package.json [hunk 2]
```diff
@@ -15,6 +15,11 @@
     "seed": "knex seed:run --specific=main.js",
     "seed-dev": "NODE_ENV=development DEV_DB_HOST=localhost npm run seed",
     "seed-users": "knex seed:run --specific=base-users.js",
+    "make-mongoose-migrate": "migrate --config mongo-migrate.js create",
```
Comment (tbolt) [inline api/package.json:18]: I'm seeing this error when I try to do either this or the `-dev` command
`make-mongoose-migrate`

File: api/package.json (hunk 1 above)
Comment (thetif) [reply api/package.json:22]: yeah that's a good idea, I'll update that and then move it back to code review

**Review by knollfear — APPROVED** [review]
LGTM!

**Pull request merged**